feat: add lined-heading class to typography

nexdrew opened the pull request:
[finishes: #119242391]

Looks like this:

<img width="910" alt="screen shot 2016-05-11 at 4 42 03 pm" src="https://cloud.githubusercontent.com/assets/1929625/15195773/5692c0f6-1797-11e6-9455-69a7843aee79.png">

Jeff did all the real work.

nexdrew committed:
chore: use vars for divider colors for reusability

nexdrew committed:
feat: add lined-heading class to typography

gabrrra commented:
@nexdrew this looks good! 
Did you make sure it works on mobile and other screen sizes?
I don't have the ability to push this up to jersey.npm.red yet, so please double check that it works responsibly and if it does, remove the needs-design-review tag and mark the story as complete.

nexdrew commented:
@jerrygabra I have not tested on multiple screen sizes. I will do so and get back to you.

nexdrew commented:
So a couple things I've noticed trying to use this in a page:
1. `.lined-heading` currently uses a 2px wide divider/line, but our current standards for dividers are either 1px (e.g. `.divider-alternate-1`) or 3px (e.g. `.divider-alternate-2`)
2. My styleguide documentation is not really conventional and should probably be cleaned up

nexdrew commented:
@jerrygabra Here are some screenshots at different grid sizes. If you can ignore the other ugliness around it, I think the lined heading works well:

![screen shot 2016-05-12 at 1 58 45 pm](https://cloud.githubusercontent.com/assets/1929625/15224988/340eac4e-184a-11e6-891f-5cab8a9b5bf7.png)

![screen shot 2016-05-12 at 1 59 03 pm](https://cloud.githubusercontent.com/assets/1929625/15224997/41c82374-184a-11e6-952f-b626db276208.png)

![screen shot 2016-05-12 at 1 59 17 pm](https://cloud.githubusercontent.com/assets/1929625/15225003/4c694984-184a-11e6-97aa-d4894b760825.png)

![screen shot 2016-05-12 at 1 59 35 pm](https://cloud.githubusercontent.com/assets/1929625/15225015/58310608-184a-11e6-910b-f1b35a0e63e2.png)

![screen shot 2016-05-12 at 1 59 45 pm](https://cloud.githubusercontent.com/assets/1929625/15225023/62588b60-184a-11e6-8b88-62f21dd0fbcf.png)

Let me know what you think.

nexdrew committed:
opt for one divider color var per color

switch unused divider-2 to be 2px

add divider-3 for 3px

cleanup lined-heading css and docs

nexdrew committed:
bump @npmcorp/pui-css-dividers to next major alpha

aredridel commented:
This is shiny!

aredridel commented on src/pivotal-ui/components/typography/typography.scss:
```diff
@@ -865,6 +865,48 @@ Headings are spaced so their line height and padding are consistent on one or ma
+    border-bottom: 2px solid $divider-alternate-color;
+    position: absolute;
+    width: 100%;
+    bottom: calc(0.5em + 2px);
```
Where does the 2px come from?

nexdrew replied:
Good question. From Jerry's design attached to the tracker story: https://s3.amazonaws.com/prod.tracker2/resource/61102449/Screen%20Shot%202016-05-10%20at%205.05.46%20PM.png?AWSAccessKeyId=AKIAJEX3ET63U5T77TYA&Expires=1463089420&Signature=wvzKkZHZ2he546GJZzkCb4Xz%2BVs%3D

aredridel replied:
Interesting -- dependent on typeface?

I also wonder if the unit we want is `1ex` -- the x-height of the typeface.

nexdrew replied:
Er, you mean in the `calc()`? That is just what Jeff and I came up with.

jefflembeck replied:
So, it might be good for us to make a variable out of that.

aredridel replied:
Indeed. Or document how it relates to the whole.

jefflembeck replied:
A. TIL about `ex` units. Whoa.

B. Maybe? we're definitely dealing in the realm of the typeface, so we might want something that is a function of that.

aredridel replied:
Yeah. Just something to think about for resilience as design changes happen. We do have the failsafe of someone looking at the design, but having it implemented in terms of why it needs to be that way then of course makes it resilient by default and we don't have to screw with it.

nexdrew replied:
@aredridel @jefflembeck Looks like `bottom: 1ex;` gives us the same result. I will change. But should this be a variable?
<img width="442" alt="screen shot 2016-05-12 at 5 24 20 pm" src="https://cloud.githubusercontent.com/assets/1929625/15230960/6ac9e930-1866-11e6-93e2-4e7838943533.png">

aredridel replied:
I suspect not -- because now it's intrinsic to the design.

nexdrew replied:
Cool, thanks!

nexdrew commented:
@aredridel Thanks! Now if I could just get Jeff to implement some more UI magic for me, I'd get the rest of the page content to be shiny too. ;)

nexdrew committed:
use 1ex based on @aredridel's review

nexdrew commented:
(BTW I'm assuming we will squash these commits when we merge. If that's not going to be the case, I will manually squash beforehand.)

aredridel commented:
We don't usually squash. Not against it particularly, but I find the history useful when figuring out why changes were made.

aredridel commented:
I see no reason not to ship this!

nexdrew commented:
@aredridel I assume I should first bump versions to non-alpha, merge, and then publish the two updated packages?

nexdrew committed:
promote dividers-3.0.0 and typography-4.2.0 to non-alpha

nexdrew commented:
@jerrygabra Any objections to this? Did you have a chance to check out the screenshots above?

gabrrra commented:
Looks good @nexdrew !
